paas_app_charmer/_gunicorn/charm.py

```diff
@@ -194,12 +183,60 @@ def restart(self) -> None:
             self._update_app_and_unit_status(
                 ops.MaintenanceStatus("Preparing service for restart")
             )
-            self._wsgi_app.restart()
+            self._build_wsgi_app().restart()
         except CharmConfigInvalidError as exc:
             self._update_app_and_unit_status(ops.BlockedStatus(exc.msg))
             return
         self._update_app_and_unit_status(ops.ActiveStatus())
 
+    def _gen_environment(self) -> dict[str, str]:
+        """Generate the environment dictionary used for the App.
+
+        This method is useful to generate the environment variables to
+        run actions against the workload container for subclasses.
+
+        Returns:
+            A dictionary representing the application environment variables.
+        """
+        return self._build_wsgi_app().gen_environment()
```

Comment:
Is this still here so that there is a hook that can override this in a charm?

Reply:
It is because the charms can use it for actions (for example the create-superuser in django). I believe this is cleaner. WDYT?